src/adjacently.rs

```diff
@@ -133,11 +136,26 @@ impl<'de, T: ?Sized> Visitor<'de> for TaggedVisitor<T> {
                         return Err(de::Error::duplicate_field(self.content));
                     }
                     // There is no second key.
-                    None => return Err(de::Error::missing_field(self.tag)),
+                    None => {
+                        if let Some(variant) = self.default_variant {
+                            // No variant is specified but there is a default variant.
+                            if let Some(Some(deserialize_fn)) = self.registry.map.get(variant) {
+                                let fn_apply = FnApply { deserialize_fn: *deserialize_fn };
+                                let content = content.into_deserializer();
+                                fn_apply.deserialize(content)?
+                            } else {
+                                // The default variant is not found in the registry.
+                                return Err(de::Error::unknown_variant(variant, &self.registry.names));
+                            }
```

Comment:
This would be simpler using the `map_lookup` already used above.

```suggestion
                            let deserialize_fn = map_lookup.visit_str(variant)?;
                            let fn_apply = FnApply { deserialize_fn };
                            let content = content.into_deserializer();
                            fn_apply.deserialize(content)?
```